### DISCUSSION TITLE
Is it reasonable to require all contributed code has godoc?

### DISCUSSION THREAD

**Discussion by geoknee** [statechannels/go-nitro#264] [General]
https://go.dev/blog/godoc

I feel like this would be a good "rule" to try and stick to. I'm not sure if there any linters or tools (other than `godoc`) that could help us stick to the rule. 

 We could add a checkbox to the PR template for this repository.

**Comment by NiloCK** [comment]
In a word: yes.

Notable:

Godocs exist also at the level of struct fields and interface methods.

![image](https://user-images.githubusercontent.com/10780590/156199109-61845492-4ffe-4198-b2e9-fa85747f9567.png)

![image](https://user-images.githubusercontent.com/10780590/156199018-b5ec52ee-651f-466d-ab6f-eb3e8340742e.png)

These comments are discoverable in-editor, but as far as I can tell are NOT exposed / exported as part of the generated package documentation.

I think a good minimum rule is that all godoc comments which WILL be output as generated documentation (comments adjacent to package-scoped identifiers) should exist.
